core/convenience.js

```diff
@@ -64,8 +64,8 @@
     rp.key = password;
 
     /* do the encryption */
-    if (p.mode === "ccm" && sjcl.arrayBuffer && sjcl.arrayBuffer.ccm && plaintext instanceof ArrayBuffer) {
-      p.ct = sjcl.arrayBuffer.ccm.encrypt(prp, plaintext, p.iv, adata, p.ts);
+    if (p.mode === "ccm" && sjcl.arrayBuffer && sjcl.arrayBuffer.ccm) {
+      p.ct = sjcl.arrayBuffer.ccm.compat_encrypt(prp, plaintext, p.iv, adata, p.ts);
```

Comment:
There is one disadvantage in this implementation in comparison to the previous way:
If an ArrayBuffer is used as input, we still return ct as a sjcl-bitArray, making it necessary to transform the ct arrayBuffer into bitArray which requires unnecessary amounts of speed/time.

Maybe we can still return an arrayBuffer if we get an arrayBuffer as input?

Also because of the padding the behaviour for arrayBuffer input also changed. I'm still not sure why the old behaviour was not valid though (why do we need padding if our input is an arrayBuffer?)

From my point of view we should not use `compat_encrypt` / `compat_decrypt` if we get passed an arrayBuffer but simply `arrayBuffer.ccm.encrypt` / `decrypt`.